### ISSUE TITLE
Uses wrong database

### ISSUE THREAD

**Issue by gildebrand** [corcel/acf#1]
I installed this package along with Corcel 1.3.0, but ran into an issue. This package seems to use a different database than Corcel uses. With Corcel we defined which database to look in manually in our `Post.php` file. How would we define which database that Corcel-ACF should use? Out of the box it seems to use the default database of the Laraval instance.

Thank you in advance!
Best regards,
Anton Gildebrand

**Comment by niknickolas** [comment]
Yes Corcel\PostMeta  uses default connection in trying to fetch field key .

**Comment by jgrossi** [comment]
@gildebrand Corcel by default uses `default` connection name. If you're using Laravel you should set up another connection and call it `corcel` or `wordpress`. Then create your own `Post` class extending the Corcel one, setting that connection:

```php
namespace App;

class Post extends \Corcel\Post {
    protected $connection = 'corcel';
}
```

Then use your own `App\Post` class, like:

```php
$post = App\Post::find(1);
echo $post->acf->text('my_text_field');
```
This should work. The `ACF` does not setup any database connection, just uses the default Corcel behaviour. Once you configured your `App\Post` class extending `Corcel\Post` with your connection, this plugin should work fine.

Let me know any issue. This plugin is under development so any issue or suggestion just advise me please.

JG

**Comment by niknickolas** [comment]
Hi, I have already read documentation of Corcel.

```
use Corcel\Post;
class WP_News extends Post
{
    protected $connection = 'wordpress';
}

```
but in this case 
```
abstract class BasicField {
public function __construct()
    {
        $this->postMeta = new PostMeta(); 
// PostMeta does not use any inheritance of current Corcel model
    }
}
```

and 

```
public function fetchFieldKey($fieldName, Post $post)
    {
        $this->post = $post;
        $this->name = $fieldName;
// in this place postMeta uses  default connection
        $postMeta = $this->postMeta->where('post_id', $post->ID)
            ->where('meta_key', '_'.$fieldName)
            ->first();

        if (!$postMeta) {
            return null;
        }

        $this->key = $postMeta->meta_value;

        return $this->key;
    }
```
after that no one of these methods of fetching data from db does not work for me
```
App\Model\WP_News::find(18)->acf->image
App\Model\WP_News::find(18)->acf->text('image')
```

Thanks for your quick answer.
Let me know if I am wrong.

**Comment by jgrossi** [comment]
@niknickolas Hm you're absolutely right! Even setting the `key` like `$post->acf->text('field')` it'll use the wrong connection. Let me think how I can do that. If you have suggestions let me know.

JG

**Comment by jgrossi** [comment]
@niknickolas BTW I started `corcel/laravel` plugin (http://github.com/corcel/laravel), that handles all Laravel stuff, so this database connection problem won't happen again. It's under development yet, and any help is appreciated. JG

**Comment by jgrossi** [comment]
@niknickolas I made some changes in the `corcel/acf` plugin. If you can, install the `develop` branch:

```
"corcel/acf": "dev-develop"
```

I did not create any phpunit test for now but I'll. Please test it and let me know if it's working.

JG

**Comment by niknickolas** [comment]
So it is better than it was.
I cannot help you fully, because of don't understand how wordpress and ACF plugin should work. I have never used wordpress before, but now I have to use laravel and wordpress as administration panel for posting news.

So, now I have other error messages 
Trying to get property of non-object in vendor/corcel/acf/src/Field/BasicField.php on line 118

```
public function fetchFieldType($fieldKey)
    {
        $post = $this->post->where('post_name', $fieldKey)->first();
        $fieldData = unserialize($post->post_content);

        $this->type = isset($fieldData['type']) ? $fieldData['type'] : 'text';

        return $this->type;
    }
```

I think that  data for this particular key  such as instances of Image has to be in wp_postmeta but not in wp_posts.

As I found out data of my key "image" is saved in wp_postmeta table, 
I suppose that there is some kind of logic error in package, that is hard to catch.

I wish you reach desired purpose.
Thanks for your attention.

**Comment by jgrossi** [comment]
@niknickolas What's the code you're calling when you get this error?

**Comment by jgrossi** [comment]
@niknickolas It seems you don't have the ACF custom fields saved in the database. Instead of using `$post->acf->image` try to use `$post->acf->image('image');`. I think it's enough. JG

**Comment by niknickolas** [comment]
@jgrossi Ok, I use that [ACF](https://www.advancedcustomfields.com/) plugin and believe that your plugin  was written to work with  WordPress  and Laravel  together,  without headache.

What are different between these two pieces of code:

in wordpress style:

```

<?php 


$image = get_field('image');
	
?>
<h1><?php the_title(); ?></h1>

<div class="hero">
	<img src="<?php echo $image['url']; ?>" />
</div>

```

in laravel style:
```

$post = App\Model\Post::find(1);

somewhere in template:

<h1>{{$post->post_title}}</h1>

<div class="hero">
	<img src="{{$post->acf->image->url}}" />
</div>

```

in wordpress style all work perfectly but in laravel style even this piece of code 
`
$post->acf->image 
`
does not work ?

**Comment by jgrossi** [comment]
@niknickolas use this instead:

```
<div class="hero">
    <img src="{{$post->acf->image('image')->url}}" />
</div>
```

The `ACF` is a new plugin and it has some problems that might be fixed soon. But this should work.

**Comment by niknickolas** [comment]
@jgrossi Hi
this way $post->acf->image('image')  works for me but not long. 
There  exists another error
```
 ErrorException in Image.php line 162: Undefined index: sizes.
```
you try to fetch ('sizes') field  from metadata 
```

fillMetadataFields(array $imageData)
    {
        $this->filename = basename($imageData['file']);
        $this->width = $imageData['width'];
        $this->height = $imageData['height'];
        $this->sizes = $imageData['sizes'];
    }
```

that are exists in the table postmeta under 
meta_key  =  _wp_attachment_metadata 

```
fetchMetadataValue(Post $attachment)
    {
        $meta = $this->postMeta->where('post_id', $attachment->ID)
            ->where('meta_key', '_wp_attachment_metadata')
            ->first();

        return unserialize($meta->meta_value);
    }

```

but in wordpress way they  use  dynamic method to retrieve all possible sizes based on wordpress themes configurations I think so, but might make mistakes regarding to wordpress themes, it can be configuration related to wordpress

**Comment by cjke** [comment]
Can confirm the dev branch correctly uses the right connection

**Comment by jgrossi** [comment]
Now it's fixed on version `0.1.3`. JG
